types/src/network_address/mod.rs

```diff
@@ -814,15 +937,16 @@ mod test {
         );
 
         let test_cases = [
+            ("/memory/1234", vec![Memory(1234)]),
             (
-                "/memory/1234/ln-handshake/0",
-                vec![Memory(1234), Handshake(0)],
-            ),
-            (
-                "/ip4/12.34.56.78/tcp/1234/ln-handshake/123",
+                &(format!(
+                    "/ip4/12.34.56.78/tcp/1234/ln-noise-ik/{}/ln-handshake/123",
+                    pubkey_str
+                )[..]),
```

Comment:
What's the `[..]` for?  You should be able to just use the reference to the format string.  If not you might have to set it to a variable first, then reference it.